app/components/courses/summary_card_component.rb

```diff
@@ -0,0 +1,226 @@
+
+      [
+        content,
+        hint
+      ].join.html_safe
+    end
+
+    def visa_sponsorship_key
+      t('.visa_sponsorship_key')
+    end
+
+    def visa_sponsorship_value
+      t(".visa_sponsorship_value.#{course.visa_sponsorship}")
+    end
+
+    private
+
+    def school_term
+      t(".location_value.school_term.#{course.funding}", default: t('.location_value.school_term.default'))
+    end
+
+    def search_by_location?
+      @search_params[:location].present?
+    end
+
+    def fee_content
+      fees = [uk_fees, international_fees].compact_blank
+
+      safe_join(fees, tag.br)
+    end
+
+    def uk_fees(fee_uk = course.enrichment_attribute(:fee_uk_eu))
+      t('.fee_value.fee.uk_fees_html', value: content_tag(:b, number_to_currency(fee_uk.to_f)))
+    end
+
+    def international_fees(fee_international = course.enrichment_attribute(:fee_international))
+      return if fee_international.blank?
+
+      t('.fee_value.fee.international_fees_html', value: content_tag(:b, number_to_currency(fee_international.to_f)))
+    end
+
+    def fee_hint
+      return nil if hide_fee_hint?
+
+      if financial_incentive.bursary_amount.present? && financial_incentive.scholarship.present?
+        t(
+          '.fee_value.fee.hint.bursaries_and_scholarship_html',
+          bursary_amount: number_to_currency(financial_incentive.bursary_amount),
+          scholarship_amount: number_to_currency(financial_incentive.scholarship)
+        )
+      elsif financial_incentive.bursary_amount.present?
+        t(
+          '.fee_value.fee.hint.bursaries_only_html',
+          bursary_amount: number_to_currency(financial_incentive.bursary_amount)
+        )
+      elsif financial_incentive.scholarship.present?
+        t(
+          '.fee_value.fee.hint.scholarship_only_html',
+          scholarship_amount: number_to_currency(financial_incentive.scholarship)
+        )
+      end
+    end
+
+    def hide_fee_hint?
+      !bursary_and_scholarship_flag_active_or_preview? || (search_by_visa_sponsorship? && (!physics? && !languages?)) || financial_incentive.blank?
+    end
+
+    def search_by_visa_sponsorship?
+      @search_params[:can_sponsor_visa].present?
+    end
+
+    PHYSICS_SUBJECT = 'Physics'
+
+    def physics?
+      main_subject&.subject_name == PHYSICS_SUBJECT
+    end
+
+    LANGUAGE_SUBJECTS = [
+      'Ancient Greek',
+      'Ancient Hebrew',
+      'English',
+      'English as a second or other language',
+      'French',
+      'German',
+      'Italian',
+      'Japanese',
+      'Latin',
+      'Mandarin',
+      'Modern Languages',
+      'Modern languages (other)',
+      'Russian',
+      'Spanish'
+    ].freeze
+
+    def languages?
+      main_subject&.subject_name.in?(LANGUAGE_SUBJECTS)
+    end
+
+    def financial_incentive
+      @financial_incentive ||= main_subject&.financial_incentive
+    end
+
+    def main_subject
+      @main_subject ||= Subject.find_by(id: course.master_subject_id)
+    end
```

Comment:
We need to create the "main subject" concept into the app.

For now I added here.

I will create a trello card and make sure it is finished before the prefilter work is delivered.